scheduler/roles/slurm/tasks/db.yml

```diff
@@ -0,0 +1,129 @@
+
+- name: Is slurm_db_password defined
+  ansible.builtin.set_fact:
+    slurm_db_password: "{{ slurm_db_password_default }}"
+  when: (slurm_db_password is not defined)
+
+- name: Install db packages
+  ansible.builtin.package:
+    name: "{{ db_packages[ansible_os_family][slurm_db_type] }}"
+    state: present 
+  register: new_install
+
+- name: Create my.cnf.d file
+  ansible.builtin.lineinfile:
+    path: /etc/my.cnf
+    line: "[client-server]"
+    create: yes
+
+- name: Include my.cnf.d directory
+  ansible.builtin.lineinfile:
+    path: /etc/my.cnf
+    insertafter: '^\[client-server\]'
+    line: '!includedir /etc/my.cnf.d'
+
+- name: Set db port
+  ansible.builtin.lineinfile:
+    path: "{{ slurm_mysql_cnf_path }}"
+    regexp: '^port=.*'
+    insertafter: '^[mysqld]'
+    line: "port={{ slurm_db_port }}"
+  when: slurm_db_type == 'mysql'
+  notify:
+    - Restart mysqld
+    - Restart slurmctld
+    - Restart slurmdbd
+
+- name: Set db port
+  ansible.builtin.lineinfile:
+    path: "{{ slurm_mariadb_cnf_path }}"
+    regexp: '^port=.*'
+    insertafter: '^[mysqld]'
+    line: "port={{ slurm_db_port }}"
+  when: slurm_db_type == 'mariadb'
+  notify:
+    - Restart mariadb
+    - Restart slurmctld
+    - Restart slurmdbd
+
+- name: Enable db port
+  ansible.posix.firewalld:
+    port: "{{ slurm_db_port }}/tcp"
+    permanent: true
+    state: enabled
+    immediate: true
+
+- name: Enable and start the mysql service
+  ansible.builtin.service:
+    name: mysqld
+    enabled: true
+    state: started
+  register: ctld_service
+  when: slurm_db_type == 'mysql'
+
+- name: Enable and start the mariadb service
+  ansible.builtin.service:
+    name: mariadb
+    enabled: true
+    state: started
+  register: ctld_service
+  when: slurm_db_type == 'mariadb'
+
+- name: Add ansible db user
+  community.mysql.mysql_user:
+    login_user: "{{ slurm_db_username_default }}"
+    login_password: "{{ slurm_db_password_default }}"
+    name: "{{ slurm_db_username }}"
+    password: "{{ slurm_db_password }}"
+    host: "localhost"
+    priv: '*.*:ALL,GRANT'
+    login_unix_socket: "{{ slurm_db_login_unix_socket }}"
+    state: present
+  when: new_install.changed
+
+- name: Add db user - shortname
+  community.mysql.mysql_user:
+    login_user: "{{ slurm_db_username }}"
+    login_password: "{{ slurm_db_password }}"
+    name: "{{ slurm_dbd_db_username }}"
+    password: "{{ slurm_dbd_db_password }}"
+    host: "{{ groups['slurm_dbd'][0] }}"
+    priv: "{{ slurm_dbd_conf_dict['StorageLoc'] }}.*:ALL,GRANT"
+    login_unix_socket: "{{ slurm_db_login_unix_socket }}"
+    state: present
+  notify:
+    - Restart slurmctld
+    - Restart slurmdbd
+
+- name: Add db user - FQDN
+  community.mysql.mysql_user:
+    login_user: "{{ slurm_db_username }}"
+    login_password: "{{ slurm_db_password }}"
+    name: "{{ slurm_dbd_db_username }}"
+    password: "{{ slurm_dbd_db_password }}"
+    host: "{{ groups['slurm_dbd'][0] }}.%"
```

Comment:
@Cypher-Miller of the short name and fqdn, if only one is essential we can remove the other. 
Please try and see if we can keep only one task.